storm-core/src/jvm/backtype/storm/grouping/PartialKeyGrouping.java

```diff
@@ -65,7 +66,11 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
                 List<Object> selectedFields = outFields.select(fields, values);
                 ByteBuffer out = ByteBuffer.allocate(selectedFields.size() * 4);
                 for (Object o: selectedFields) {
-                    out.putInt(o.hashCode());
+                    if (o instanceof Object[]) {
```

Comment:
I see. Yes, I could add this.  However, since we do not currently do this in backtype.storm.tuple it would be inconsistent.

In b.s.tuple, I always get a List<Object>.  I convert it to an array there because the call to Arrays#deepHashCode is simple.  If a tuple is _itself_ a List, then Arrays#hashCode will call List#hashCode on it.